File: 1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		ans := target - num
		idx, ok := m[ans]
		if ok {
			return []int{idx, i}
		}
		m[num] = i
	}
	return nil
}

File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {}
        for i in range(len(nums)):
            ans = target - nums[i]
            if ans in lookup:
                return [lookup[ans], i]
            lookup[nums[i]] = i
        

